File: backend/src/main/java/com/school/saas/module/course/dto/CourseDTO.java
package com.school.saas.module.course.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CourseDTO {
    private UUID id;
    private UUID schoolId;
    private UUID classRoomId;
    private String classRoomName;
    private UUID teacherId;
    private String teacherName;
    private UUID specialityId;
    private String specialityName;
    private String specialityCode;
    private String subject;
    private String subjectCode;
    private String description;
    private String schedule;
    private String semester;
    private String documents;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: mobile/lib/models/course_dto.dart
class CourseDTO {
  final String id;
  final String schoolId;
  final String classRoomId;
  final String classRoomName;
  final String teacherId;
  final String teacherName;
  final String specialityId;
  final String specialityName;
  final String specialityCode;
  final String subject;
  final String subjectCode;
  final String? description;
  final String schedule;
  final String semester;
  final String createdAt;
  final String updatedAt;

  CourseDTO({
    required this.id,
    required this.schoolId,
    required this.classRoomId,
    required this.classRoomName,
    required this.teacherId,
    required this.teacherName,
    required this.specialityId,
    required this.specialityName,
    required this.specialityCode,
    required this.subject,
    required this.subjectCode,
    this.description,
    required this.schedule,
    required this.semester,
    required this.createdAt,
    required this.updatedAt,
  });

  factory CourseDTO.fromJson(Map<String, dynamic> json) {
    return CourseDTO(
      id: json['id'],
      schoolId: json['schoolId'],
      classRoomId: json['classRoomId'],
      classRoomName: json['classRoomName'],
      teacherId: json['teacherId'],
      teacherName: json['teacherName'],
      specialityId: json['specialityId'],
      specialityName: json['specialityName'],
      specialityCode: json['specialityCode'],
      subject: json['subject'],
      subjectCode: json['subjectCode'],
      description: json['description'],
      schedule: json['schedule'],
      semester: json['semester'],
      createdAt: json['createdAt'],
      updatedAt: json['updatedAt'],
    );
  }
}
